File: utils/url.go
package utils

import "net/url"

func IsValidUrl(urlStr string) bool {
	if len(urlStr) > 2048 {
		return false
	}
	urlObj, err := url.ParseRequestURI(urlStr)
	return err == nil && (urlObj.Scheme == "http" || urlObj.Scheme == "https") && urlObj.Host != ""
}

File: web/src/utils/url.ts
export function isValidUrl(url: string): boolean {
	if (url.length > 2048) {
		return false;
	}
	let urlObj;
	try {
		urlObj = new URL(url);
	} catch {
		return false;
	}
	return urlObj.protocol === "http:" || urlObj.protocol === "https:";
}
